From 5633d284a6c77882487ef58885dbcbfd24c07f9c Mon Sep 17 00:00:00 2001
From: hvadehra <hvd@google.com>
Date: Fri, 11 Aug 2023 09:10:16 +0200
Subject: [PATCH] Migrate usages deprecated `JavaInfo` fields

transitive_deps was an alias for transitive_compile_time_jars transitive_runtime_deps was an alias for transitive_runtime_jars

The fields were deprecated in 2021, and are dropped in Bazel@HEAD

Fixes bazelbuild#1003
---
 kotlin/internal/jvm/compile.bzl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kotlin/internal/jvm/compile.bzl b/kotlin/internal/jvm/compile.bzl
index 80cbf7a8..327a0bdc 100644
--- a/kotlin/internal/jvm/compile.bzl
+++ b/kotlin/internal/jvm/compile.bzl
@@ -261,7 +261,7 @@ def _run_merge_jdeps_action(ctx, toolchains, jdeps, outputs, deps):
     )
 
     # For sandboxing to work, and for this action to be deterministic, the compile jars need to be passed as inputs
-    inputs = depset(jdeps, transitive = [depset([], transitive = [dep.transitive_deps for dep in deps])])
+    inputs = depset(jdeps, transitive = [depset([], transitive = [dep.transitive_compile_time_jars for dep in deps])])
 
     ctx.actions.run(
         mnemonic = mnemonic,
